tests/rkt-monitor/main.go

```diff
@@ -0,0 +1,296 @@
+
+		for _, ps := range usage {
+			usages[ps.Pid] = append(usages[ps.Pid], ps)
+		}
+
+		_, err = process.NewProcess(int32(execCmd.Process.Pid))
+		if err != nil {
+			// process.Process.IsRunning is not implemented yet
+			fmt.Fprintf(os.Stderr, "rkt exited prematurely\n")
+			break
+		}
+
+		time.Sleep(time.Second)
+	}
+
+	err = killAllChildren(int32(execCmd.Process.Pid))
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "cleanup failed: %v\n", err)
+	}
+
+	for _, processHistory := range usages {
+		var avgCPU float64
+		var avgMem uint64
+		var peakMem uint64
+
+		for _, p := range processHistory {
+			avgCPU += p.CPU
+			avgMem += p.RSS
+			if peakMem < p.RSS {
+				peakMem = p.RSS
+			}
+		}
+
+		avgCPU = avgCPU / float64(len(processHistory))
+		avgMem = avgMem / uint64(len(processHistory))
+
+		fmt.Printf("%s(%d): seconds alive: %d  avg CPU: %f%%  avg Mem: %s  peak Mem: %s\n", processHistory[0].Name, processHistory[0].Pid, len(processHistory), avgCPU, formatSize(avgMem), formatSize(peakMem))
+	}
+}
+
+func killAllChildren(pid int32) error {
+	p, err := process.NewProcess(pid)
+	if err != nil {
+		return err
+	}
+	processes := []*process.Process{p}
+	for i := 0; i < len(processes); i++ {
+		children, err := processes[i].Children()
+		if err != nil {
+			if exitErr, ok := err.(*exec.ExitError); ok && exitErr.Sys().(syscall.WaitStatus).ExitStatus() == 1 {
+				// An ExitError with a code of 1 will be returned when there are no children
+				continue
+			} else {
+				return err
+			}
+		}
+		processes = append(processes, children...)
+	}
+	for _, p := range processes {
+		osProcess, err := os.FindProcess(int(p.Pid))
+		if err != nil {
+			if err.Error() == "os: process already finished" {
+				continue
+			}
+			return err
+		}
+		err = osProcess.Kill()
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func getUsage(pid int32) ([]*ProcessStatus, error) {
+	var statuses []*ProcessStatus
+	pids := []int32{pid}
+	for i := 0; i < len(pids); i++ {
+		proc, ok := pidMap[pids[i]]
+		if !ok {
+			var err error
+			proc, err = process.NewProcess(pids[i])
+			if err != nil {
+				return nil, err
+			}
+			pidMap[pids[i]] = proc
+		}
+		s, err := getProcStatus(proc)
+		if err != nil {
+			return nil, err
+		}
+		statuses = append(statuses, s)
+
+		children, err := proc.Children()
+		if err != nil {
+			if exitErr, ok := err.(*exec.ExitError); ok && exitErr.Sys().(syscall.WaitStatus).ExitStatus() == 1 {
+				// An ExitError with a code of 1 will be returned when there are no children
+				continue
+			} else {
+				return nil, err
+			}
+		}
+
+	childloop:
+		for _, child := range children {
+			for _, p := range pids {
+				if p == child.Pid {
+					fmt.Printf("%d is in %#v\n", p, pids)
+					continue childloop
+				}
+			}
+			pids = append(pids, child.Pid)
+		}
+	}
+	return statuses, nil
+}
+
+func getProcStatus(p *process.Process) (*ProcessStatus, error) {
+	n, err := p.Name()
+	if err != nil {
+		return nil, err
+	}
+	c, err := p.CPUPercent(0)
```

Comment:
It seems this doesn't exist anymore in https://github.com/shirou/gopsutil/blob/master/process/process.go

Also, we should vendor it.